### PR TITLE
Put states to entries

### LINKED ISSUE #3: assign states to each entry
A key objective of this project is to split out the news entries by state, and eventually by city.

To do this, lets make use of the `locationtagger` library: https://pypi.org/project/locationtagger/ and check against the descriptions and link URLs. 

If the state isn't named directly, we can create a mapping of publications to states/cities and assign them based on that

### PR DESCRIPTION
**Pull request by SuragNuthulapaty**
# Overview

Fixes #3 

Gets a state and/or city related to each blurb and adds it as a new column in the table.

# Notes

Not all rows have a location attached to it due to how the library works. We might need to go in manually and fix up everything else.

### CODE REVIEW

File: scrape.py [hunk 1]
```diff
@@ -46,10 +47,19 @@ def bulletPointScrape(arr, link, date, category):
     curItem['blurb'] = blurbText
     curItem['links-within-blurb'] = links
     curItem['article-link'] = link
+    curItem['states'] = getStates(curItem).regions
+    curItem['cities'] = getStates(curItem).cities
```
Comment (derekeder) [inline scrape.py:51]: the array brackets `[` make this a little harder to work with in the spreadsheet. lets convert them into a string with a comma separated list. 

We can do something like this:
`curItem['states'] = ",".join(str(x) for x in curItem['states'])`

Lets do this for, `cities`, `regions` and `links-within-blurb`

**Review by derekeder — CHANGES_REQUESTED** [review]
@SuragNuthulapaty Looking good. I have one comment on formatting and some notes on the accuracy and ways to improve our coverage.

I put all the latest data into a new sheet and did a summary pivot table: https://docs.google.com/spreadsheets/d/1SZ1NrJgAnjUwHo_ZVv-TwRrb1-wdb64wVLcIkMI_XiU/edit#gid=1630528016

A few observations for the [State classifications](https://docs.google.com/spreadsheets/d/1SZ1NrJgAnjUwHo_ZVv-TwRrb1-wdb64wVLcIkMI_XiU/edit#gid=1525064602):
* 35,500 / 90,993 have no state assigned (61% coverage)
* it does seem to be missing some obvious ones like `Hawaii’s only coal plant is on track to shut down by Sept. 1 ...` and `The ongoing drought in the Colorado River Basin ...`. I wonder if accuracy would improve if we didn't concatenate the link and just went with blurb & publication.
* some of the state classifications are inaccurate (seeing `North`, `West` and `Paris` as values). we should run the found states through a known list of valid states and filter out the incorrect ones
* many entries have more than one state assigned, which is ok

A few observations for the [City classifications](https://docs.google.com/spreadsheets/d/1SZ1NrJgAnjUwHo_ZVv-TwRrb1-wdb64wVLcIkMI_XiU/edit#gid=935733072):
* 39,076 / 90,993 have no city assigned (57% coverage)
* City classification is much less accurate. Seeing State names and other words like `Hill` `Energy` and `House`. This is harder to clean up as it would require a list of all possible cities in the US.

**Comment by derekeder** [comment]
@SuragNuthulapaty pulled in the latest scrape from yesterday.

Looks like we successfully removed most of the false positives. However, our overall tagging coverage went down:

38,759 / 93,489 untagged = 58.5% coverage

I think this is good to bring in. We can work on improving the accuracy in a future PR

**Comment by SuragNuthulapaty** [comment]
@derekeder I just pushed up the newest run, it should be up to date now

**Review by derekeder — APPROVED** [review]
@SuragNuthulapaty cool. looks like the digestItems didn't change from the one I used earlier. this looks good to bring in!

**Pull request merged**